packages/slice-machine/server/src/api/screenshots/generate.ts

```diff
@@ -59,37 +59,40 @@ export async function generateScreenshot(
     (v: Models.VariationAsObject) => v.id
   );
 
-  const promises: Promise<ScreenshotUI | Error>[] = variationIds.map(
+  const promises: Promise<ScreenshotUI>[] = variationIds.map(
     (id: Models.VariationAsObject["id"]) =>
       generateForVariation(env, libraryName, slice, id)
   );
-  const results: (ScreenshotUI | Error)[] = await Promise.all(promises);
+
+  const results = await Promise.allSettled(promises);
 
   return results.reduce(
     (
       acc: ScreenshotResults,
-      screenshotOrError: ScreenshotUI | Error,
+      result: PromiseSettledResult<ScreenshotUI>,
       index: number
     ) => {
-      if (screenshotOrError instanceof Error) {
-        return {
-          screenshots: acc.screenshots,
-          failure: [
-            ...acc.failure,
-            { error: screenshotOrError, variationId: variationIds[index] },
-          ],
-        };
-      }
-
-      return {
-        screenshots: {
-          ...acc.screenshots,
-          [variationIds[index]]: screenshotOrError,
-        },
-        failure: acc.failure,
-      };
+      const key = variationIds[index];
+
+      const screenshots =
+        result.status === "fulfilled"
+          ? {
+              ...acc.screenshots,
+              [key]: result.value,
+            }
+          : acc.screenshots;
+
+      const failure =
+        result.status === "rejected"
+          ? [
+              ...acc.failure,
+              { error: result.reason as Error, variationId: key },
```

Comment:
One thing i'm unsure of, what the type of reason, at first it's undefined so how are you sure it's an error ?
Maybe we should verify it's an error and if not instantiate one ?

That one is good example of "expected failure" haha, but letting it throw to handle it afterward was a good call 👍

Reply:
throw/catch has the signature `catch (e: any)`, I'm pretty sure it's an error from the previous code that was there

Reply:
As long as we don't throw ourself then i guess you're right :)